File: product-of-array-except-self/Yiseull.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        answer = [1]

        # answer[i] -> nums[i] 왼쪽 값들의 곱
        for i in range(1, n):
            answer.append(answer[i - 1] * nums[i - 1])

        tmp = 1
        for i in range(n - 1, -1, -1):
            answer[i] *= tmp
            tmp *= nums[i]

        return answer

File: maximum-subarray/Yiseull.java
class Solution {
    public int maxSubArray(int[] nums) {
        int answer = -10001, tmp = 0;

        for (int i = 0; i < nums.length; i++) {
            tmp += nums[i];
            answer = Math.max(answer, tmp);
            tmp = Math.max(tmp, 0);
        }

        return answer;
    }
}
